hnn_core/tests/test_dipole.py

```diff
@@ -124,17 +124,18 @@ def test_dipole_simulation():
                         record_isoma=0)
 
     # test Network.copy() returns 'bare' network after simulating
-    dpl = simulate_dipole(net, tstop=25., n_trials=1)[0]
-    net_copy = net.copy()
-    assert len(net_copy.external_drives['evprox1']['events']) == 0
-
-    # test that Dipole.copy() returns the expected exact copy
-    assert_allclose(dpl.data['agg'], dpl.copy().data['agg'])
-
-    # Test raster plot with no spikes
-    net = Network(params)
-    simulate_dipole(net, tstop=0.1, n_trials=1)
-    net.cell_response.plot_spikes_raster()
+    with pytest.raises(Warning, match='No connections'):
+        dpl = simulate_dipole(net, tstop=25., n_trials=1)[0]
+        net_copy = net.copy()
+        assert len(net_copy.external_drives['evprox1']['events']) == 0
+
+        # test that Dipole.copy() returns the expected exact copy
+        assert_allclose(dpl.data['agg'], dpl.copy().data['agg'])
+
+        # Test raster plot with no spikes
+        net = Network(params)
+        simulate_dipole(net, tstop=0.1, n_trials=1)
+        net.cell_response.plot_spikes_raster()
```

Comment:
Nitpick: which line actually returns the warning? If it is both instances of `simulate_dipole()`, should this be broken into 2 context managers that each raise the same warning?